metaflow/datatools/s3.py

```diff
@@ -540,6 +540,8 @@ def _one_boto_op(self, op, url):
                 op(s3, tmp.name)
                 return tmp.name
             except ClientError as err:
+                if 'NoSuchBucket' in str(err):
+                    raise
```

Comment:
instead, you can trap the error after L:545
```
if error_code = 'NoSuchBucket':
    raise MetaflowS3URLException("Specified S3 bucket doesn't exist.")
```

Reply:
Changes made and tested.
```
2020-02-06 11:25:12.480 [1581009911359453/start/1 (pid 30260)] S3 object not found:
2020-02-06 11:25:12.480 [1581009911359453/start/1 (pid 30260)] s3://<redacted>/data.json
2020-02-06 11:25:12.550 [1581009911359453/start/1 (pid 30260)] 
2020-02-06 11:25:12.553 [1581009911359453/start/1 (pid 30260)] Task failed.